README.md

```diff
@@ -63,178 +72,366 @@ bambuAnnotations <- prepareAnnotations(gtf.file)
+
+
+### Training a model on another species/dataset and applying it
+
+In situations where training is not or cannot be performed, and the default model is also not suitable for the sample (the sample is generated from a different technology, species, condition, etc), bambu provides the option to train a new model, if well annotated similar data is available. For example one might train a model on arabidopsis to apply to an unannotated plant sample.
 
 ```rscript
-plotBambu(se, type = "heatmap", group.var) # heatmap 
+# first train the model using a related annotated dataset from .bam
+se = bambu(reads = sample1.bam, annotations = annotations, genome = fa.file, opt.discovery = list(returnModel = FALSE))
+newDefaultModel = metadata(se)$model
 
-plotBambu(se, type = "pca", group.var) # PCA visualization
+# alternatively train the model using an rcFile
+rcFile <- readRDS(pathToRcFile)
+newDefaultModel = trainBambu(rcFile)
+
+# use the trained model on another sample
+# sample2.bam and fa.file2 represent the aligned reads and genome for the poorly annotated sample
+se <- bambu(reads = sample2.bam, annotations = NULL, genome = fa.file2, opt.discovery = list(defaultModels = newDefaultModel, fitReadClassModel = FALSE))
+
+#trainBambu Arguments
+
+rcFile <- NULL, min.readCount = 2, nrounds = 50, NDR.threshold = 0.1, verbose = TRUE
 ```
 
-**Write bambu outputs to files**
+|arguments|description|
+|---|---|
+|rcFile|A path to the intermediate rcFile sample or the loaded rcFile sample (see [Storing and using preprocessed files (rcFiles)](#Storing-and-using-preprocessed-files-rcFiles|
+|min.readCount|The minimum read count threshold used for read classes during training|
+|nroundsoutput|The number of stumps used in the xgboost tree|
+|NDR.threshold|The NDR threshold that will be used for the recommended NDR calibration when using this model. |
+|verbose|A logical if more information should be printed whilst the function is running|
+
+
+### Including single exons
 
-- ***writeBambuOutput*** will generate three files, including a *.gtf* file for the extended annotations, and two *.txt* files for the expression counts at transcript and gene levels.
+By default *bambu* does not report single exon transcripts because they are known to have a high frequency of false positives and do not have splice junctions that are used by *bambu* to distinguish read classes. Nevertheless *bambu* trains a separate model on single-exon transcripts, and these predictions can be accessed and included in the annotations.
 
 ```rscript
-writeBambuOutput(se, path = "./bambu/")
+se <- bambu(reads = sample1.bam, annotations = annotations, genome = fa.file, opt.discovery = list(min.txScore.singleExon = 0))
+```
+### Bambu Arguments
+
+|argument|description|
+|---|---|
+|reads|A string or a vector of strings specifying the paths of bam files for genomic alignments, or a BamFile object or a BamFileList object (see Rsamtools).|
```

Comment:
do you want to provide a link for Rsamtools?

Reply:
no I don't want to have to link externally which could break. I will just change this to saying frmo Rsamtools, if they have a BamFileLIst they already should know what it is, and if they don't they don't need to know